tianshou/data/utils/segtree.py

```diff
@@ -0,0 +1,87 @@
+import numpy as np
+from typing import Union, Optional
+
+
+class SegmentTree:
```

Comment:
If this needs to be as fast as possible, consider using `numpy.jit`. You may get a significant speed-up almost for free.

Reply:
It is `numba.jit(nopython=True)` instead of `np.jit`. I'm considering that we can use numba to rewrite some part of our codebase later because it is not a good deal to use numba only in this SegmentTree.